src/app/core/admin/admin-entity/admin-entity-general-settings/admin-entity-general-settings.component.ts

```diff
@@ -38,19 +48,36 @@ import { StringDatatype } from "../../../basic-datatypes/string/string.datatype"
     FontAwesomeModule,
     MatTooltipModule,
     BasicAutocompleteComponent,
+    MatCheckboxModule,
+    MatTableModule,
+    MatOptionModule,
+    MatSelectModule,
+    CommonModule,
+    MatTooltipModule,
+    HelpButtonComponent,
+    MatSort,
+    EntityFieldLabelComponent,
   ],
 })
 export class AdminEntityGeneralSettingsComponent implements OnInit {
   @Input() entityConstructor: EntityConstructor;
   @Output() generalSettingsChange: EventEmitter<EntityConfig> =
     new EventEmitter<EntityConfig>();
   @Input() generalSettings: EntityConfig;
+  allPIIFields: any[];
+  @Input() showPIIDetails: boolean;
+  entitySchemaField: EntitySchemaField;
+  dataSource: MatTableDataSource<any>;
+  anonymizOptionList: string[] = ["Retain", "Partially Anonymize", "Remove"];
```

Comment:
Let's add the tooltips here: `anonymizOptionList: { label: string; tooltip: string }[] =`
(calling methods in the template is not ideal, as they are executed on every change detection cycle of angular. And it's easier to maintain the texts, if they are all in one place here).

Reply:
i.e. remove the getTooltip method for this then

Reply:
actually, we can also hard code the three options including tooltip directly in the template html. Anyway these are not dynamically generated currently.